src/wing/CCPACSWingCell.cpp

```diff
@@ -407,246 +444,325 @@ TopoDS_Shape CCPACSWingCell::GetSkinGeometry(TiglCoordinateSystem cs) const
-                          .GetSparGeometry(WING_COORDINATE_SYSTEM);
-    }
+    auto le_intersect = GetFaceAndUV(loftShape, le_point_proj);
+    auto te_intersect = GetFaceAndUV(loftShape, te_point_proj);
 
-    double u05 = 0.5, u1 = 1., v0 = 0., v1 = 1.;
-    if (m_positioningInnerBorder.GetInputType() == CCPACSWingCellPositionSpanwise::InputType::Rib) {
-        planeShapeIB = GetRibCutGeometry(m_positioningInnerBorder.GetRib());
-    }
-    else if (IBEta1 == 0. && IBEta2 == 0.) {
-        // if the inner border of the cell is the inner border of the Component segment
-        // a cutting plane from the inner border of the WCS is created
-        // this is necessary due to cutting precision
-        BRepAdaptor_Surface surf(wsr.GetInnerFace());
-        gp_Pnt p0 = surf.Value(u05, v0);
-        gp_Pnt pU = surf.Value(u05, v1);
-        gp_Pnt pV = surf.Value(u1, v0);
-        gp_Ax3 ax0UV(p0, gp_Vec(p0, pU) ^ gp_Vec(p0, pV), gp_Vec(p0, pU));
+    gp_Vec te_to_le = gp_Vec(le_point, te_point).Normalized();
+    gp_Ax3 border_axis(le_point, zRefDir ^ te_to_le, te_to_le);
 
-        planeShapeIB = BRepBuilderAPI_MakeFace(gp_Pln(ax0UV)).Face();
+    if ( border == SpanWiseBorder::Outer ){
+        border_axis.ZReverse();
+        cache.border_outer_ax3 = border_axis;
     }
+    else {
+        cache.border_inner_ax3 = border_axis;
+    }
+
+    TopoDS_Shape result;
+    if ( te_intersect.face.IsEqual(le_intersect.face) && fabs(te_intersect.v - le_intersect.v ) < tol ){
+        // border runs along an isocurve of a single fac => we can trim
+
+        // trim along v
+        double v = le_intersect.v;
+        TopoDS_Face face = le_intersect.face;
+        Handle(Geom_Surface) surf = BRep_Tool::Surface(face);
+        Standard_Real umin, umax, vmin, vmax;
+        BRepTools::UVBounds(face, umin, umax, vmin, vmax);
+        Handle(Geom_BSplineSurface) trimmed_surf;
+        if ( border == SpanWiseBorder::Inner ) {
+            trimmed_surf = fabs(v - vmin) < tol ? GeomConvert::SurfaceToBSplineSurface(surf)
+                                                : CTiglBSplineAlgorithms::trimSurface(surf, umin, umax, v, vmax);
+        }
+        else {
+            trimmed_surf = fabs(v - vmax) < tol ? GeomConvert::SurfaceToBSplineSurface(surf)
+                                                : CTiglBSplineAlgorithms::trimSurface(surf, umin, umax, vmin, v);
+        }
+        TopoDS_Face trimmed_face = BRepBuilderAPI_MakeFace(trimmed_surf, Precision::Confusion());
+
+        // create a compound of all faces of the loft shape, where the original face
+        // gets replaced by the trimmed face
+        TopoDS_Builder builder;
+        TopoDS_Compound compound;
+        builder.MakeCompound(compound);
+        builder.Add(compound, trimmed_face);
+
+        TopTools_IndexedMapOfShape faceMap;
+        TopExp::MapShapes(loftShape, TopAbs_FACE, faceMap);
+        for (int f = 1; f <= faceMap.Extent(); f++) {
```

Comment:
If I understand the code correctly, you keep all faces except the one that is trimmed.

What I don't yet understand is the following: Lets assume you have an outer border and you are trimming a face of the first segment. Shouldn't all faces from the second segment be discarded as well? In the end it seems to me that you end up with more than just the cell.

Reply:
Actually here I just replace the original face with the trimmed face and keep all other faces. 

Starting in Line 578 all faces to one side of the border get discarded. This way I don't have to repeat the code block starting in line 578 for both the trimming and the cutting algorithm.

Reply:
Ah... I see. Maybe add some high level comment to the code block starting add line 578.